packages/mco/components/dr-status-popover/parsers/subscription-parser.tsx

```diff
@@ -0,0 +1,228 @@
+  drStatusList: {
+    healthStatus: VolumeReplicationHealth;
+  }[]
+): VolumeReplicationHealth => {
+  return drStatusList.reduce((acc: VolumeReplicationHealth, health) => {
+    if (health.healthStatus === VolumeReplicationHealth.CRITICAL) {
+      return VolumeReplicationHealth.CRITICAL;
+    } else if (
+      health.healthStatus === VolumeReplicationHealth.WARNING &&
+      acc !== VolumeReplicationHealth.CRITICAL
+    ) {
+      return VolumeReplicationHealth.WARNING;
+    }
+    return acc;
+  }, VolumeReplicationHealth.HEALTHY);
+};
+
+const parseDRStatusForGroup = (
+  group: SubscriptionGroupType
+): {
+  drPolicyName: string;
+  schedulingInterval?: string;
+  primaryCluster: string;
+  targetCluster?: string;
+  lastGroupSyncTime: string;
+  healthStatus: VolumeReplicationHealth;
+  phase: string;
+} | null => {
+  const drpc = group.drInfo?.drPlacementControl;
+  if (!drpc) return null;
+
+  const drPolicyName = getDRPolicyName(drpc);
+  const schedulingInterval = group.drInfo?.drPolicy?.spec?.schedulingInterval;
+
+  const primaryClusterName = getLastAppDeploymentClusterName(drpc);
+  const drClusters = group.drInfo?.drClusters;
+  if (!drClusters) {
+    return null;
+  }
+  const targetCluster = findCluster(drClusters, primaryClusterName);
+
+  const lastGroupSyncTime: string = drpc?.status?.lastGroupSyncTime;
+  const replicationType = getReplicationType(schedulingInterval);
+
+  const healthStatus = getReplicationHealth(
+    lastGroupSyncTime,
+    schedulingInterval,
+    replicationType
+  );
+
+  return {
+    drPolicyName,
+    schedulingInterval,
+    primaryCluster: primaryClusterName,
+    targetCluster: targetCluster?.metadata?.name,
+    lastGroupSyncTime,
+    healthStatus,
+    phase: drpc?.status?.phase,
+  };
+};
+
+const getDRResources = (
+  namespace: string
+): {
+  resources: {
+    drPolicies: ReturnType<typeof getDRPolicyResourceObj>;
+    drClusters: ReturnType<typeof getDRClusterResourceObj>;
+    drPlacementControls: ReturnType<typeof getDRPlacementControlResourceObj>;
+  };
+} => ({
+  resources: {
+    drPolicies: getDRPolicyResourceObj(),
+    drClusters: getDRClusterResourceObj(),
+    drPlacementControls: getDRPlacementControlResourceObj({
+      namespace,
+    }),
+  },
+});
+
+const getSubscriptionResources = (
+  application: ApplicationKind,
+  namespace: string,
+  drResources: DisasterRecoveryResourceKind,
+  drLoaded: boolean,
+  drLoadError: any
+): {
+  resources: {
+    subscriptions: ReturnType<typeof getSubscriptionResourceObj>;
+    placementRules: ReturnType<typeof getPlacementRuleResourceObj>;
+    placements: ReturnType<typeof getPlacementResourceObj>;
+    placementDecisions: ReturnType<typeof getPlacementDecisionsResourceObj>;
+  };
+  drResources: {
+    data: DisasterRecoveryResourceKind;
+    loaded: boolean;
+    loadError: any;
+  };
+  overrides: {
+    applications: {
+      data: ApplicationKind;
+      loaded: boolean;
+      loadError: string;
+    };
+  };
+} => ({
```

Comment:
```suggestion
): WatchResources['resources']
} => ({
```
import from subscription.ts hooks